onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_support_checker.cc

```diff
@@ -281,21 +505,29 @@ int BinaryOpSupportChecker::GetMinSupportedOpSet(const NodeUnit& node_unit) cons
   const auto& op(node_unit.OpType());
 
   // Add/Sub/Mul/Div/Pow opset 6- has broadcast attributes we do not support now
-  if (op != "QLinearAdd")
+  if (op != "QLinearAdd" && op != "QLinearMul")
     return 7;
 
   return 1;
 }
 
-bool BinaryOpSupportChecker::HasSupportedInputsImpl(const NodeUnit& node_unit) const {
-  bool is_qlinear_add = node_unit.OpType() == "QLinearAdd";
+bool BinaryOpSupportChecker::HasSupportedInputOutputsImpl(
+    const InitializedTensorSet& initializers, const NodeUnit& node_unit,
+    const OpSupportCheckParams& params) const {
+  bool is_quantized_op = IsQuantizedOp(node_unit);
   bool is_pow = node_unit.OpType() == "Pow";
-  if (!is_qlinear_add && !is_pow)
-    return BaseOpSupportChecker::HasSupportedInputsImpl(node_unit);
+  if (!is_quantized_op && !is_pow)
+    return BaseOpSupportChecker::HasSupportedInputOutputsImpl(initializers, node_unit, params);
+
+  if (is_quantized_op) {
+    // QLinearAdd/QDQAdd/QLinearMul/QDQMul
+    if (!HasValidBinaryOpQuantizedInputTypes(node_unit))
+      return false;
 
-  if (is_qlinear_add) {
-    // QLinearAdd
-    if (!HasValidBinaryOpQuantizedInputs(node_unit))
+    if (!IsQuantizedIOSupported(initializers, node_unit, {0, 1}, params, true /* is_input */))
```

Comment:
>/* is_input */ [](http://example.com/codeflow?start=78&length=14)

just a thought - if it was an enum, say `enum class IOKind { Input, Output }`, we wouldn't need comments